server/crates/server-api/src/api.rs

```diff
@@ -0,0 +1,38 @@
+//TODO: Maybe include any kind of rest apis or w/e in here, or we could go with a different
+//approach I was just offering this as a sort of skeleton setup
+
+use std::{
+    //Prelude is for including many commonly used traits in standard library, it is common
+    //convention in public crates as well.
+    io::prelude::*,
+    net::TcpStream
+};
+
+//I am just statically embeddeding the index file into the executable, it's really small for now so
+//it's dumb to block on io for it. Obviously when we make a more complicated file we may need
+//dynamic file I/O!
+const INDEX_FILE : &'static str = include_str!("../res/index.html");
+const ERROR_404_FILE : &'static str = include_str!("../res/404.html");
```

Comment:
Another neat bit about this. The file bytes gets loaded into program executable memory, so when we go to read it will be already in memory at a very localized location in physical memory relative to the instruction pointer meaning accessing it will be pretty damn fast.